src/transformState.js

```diff
@@ -4,8 +4,31 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const objects of actions) {
+    const { type, extraData, keysToRemove } = objects;
+
+    switch (type) {
+      case 'addProperties' :
+        Object.assign(state, extraData);
+        break;
+
+      case 'removeProperties' :
+        for (const deleteValue of keysToRemove) {
+          delete state[deleteValue];
+        }
+        break;
+
+      case 'clear' :
+        for (const deleteAll in state) {
+          delete state[deleteAll];
+        }
+        break;
+    }
```

Comment:
switch/case should always have default case for error handling
```suggestion
    default:
        throw new Error('Wrong type!');
    }
```